packages/teleport/src/components/LabelSelector/LabelSelector.tsx

```diff
@@ -0,0 +1,269 @@
+      </LabelContainer>
+      {showAdd && (
+        <AddLabelContainer
+          ref={addLabelContainerRef}
+          data-testid="add-label-container"
+        >
+          <AddLabelInput
+            name="addLabel"
+            value={newLabel}
+            onChange={e => {
+              setNewLabel(e.target.value);
+            }}
+            onKeyPress={e => {
+              // Add a new label on `Enter` if it's valid.
+              if (e.charCode === 13 && validLabel) {
+                handleAddLabel();
+              }
+            }}
+            ref={addLabelInputRef}
+          />
+          {validLabel ? (
+            <CreateLabel
+              onClick={handleAddLabel}
+              data-testid="create-label-msg"
+            >
+              + Create new label "{newLabel}"
+            </CreateLabel>
+          ) : (
+            <CreateLabelError data-testid="create-label-error">
+              <WarningIconWrapper>
+                <Warning />
+              </WarningIconWrapper>
+              <WarningText>
+                <Text style={{ color: '#D83C31', fontWeight: 700 }}>
+                  Invalid label format
+                </Text>
+                <Text>Follow `key:pair` format to add a new label</Text>
+              </WarningText>
+            </CreateLabelError>
+          )}
+        </AddLabelContainer>
+      )}
+    </div>
+  );
+}
+
+const Heading = styled.div`
+  clear: both;
+  height: 1.5rem;
+  position: relative;
+`;
+
+const LabelContainer = styled.div`
+  border-radius: 4px;
+  border: 1px solid rgba(255, 255, 255, 0.1);
+  box-shadow: 0px 8px 10px rgba(12, 12, 14, 0.07);
+  clear: both;
+  cursor: pointer;
+  display: flex;
+  flex-wrap: wrap;
+  gap: 10px;
+  margin-top: 8px;
+  min-height: 36px;
+  padding: 10px 16px;
+`;
+
+const AddLabelContainer = styled.div`
+  background: #182250;
+  border-radius: 4px;
+  height: 100px;
+  padding: 1rem;
+`;
+
+const AddLabelInput = styled.input`
+  background: #182250;
+  border-radius: 52px;
+  border: 1.5px solid #512fc9;
+  color: white;
+  height: 40px;
+  padding: 0 12px;
+  width: calc(100% - 2rem);
+`;
+
+const CreateLabel = styled.button`
+  background: none;
+  border: none;
+  color: white;
+  cursor: pointer;
+  font-size: 1rem;
+  margin-left: 16px;
+  margin-top: 25px;
+`;
+
+const CreateLabelError = styled.div`
+  margin-top: 8px;
+`;
+
+const WarningIconWrapper = styled.div`
+  background: rgba(255, 255, 255, 0.05);
+  border-radius: 54px;
+  float: left;
+  height: 20px;
+  line-height: 20px;
+  margin-top: 10px;
+  padding: 5px;
+  text-align: center;
+  width: 20px;
+`;
+
+const WarningText = styled.div`
+  float: left;
+  margin-left: 8px;
+`;
+
+type LabelSelectorProps = {
+  onChange: (labels: string[]) => void;
+};
+
+function labelList({
+  labels,
+  onDismiss,
+}: {
+  labels: string[];
+  onDismiss: (string) => void;
+}) {
+  return labels.map(label => (
+    <Pill key={label} label={label} dismissable onDismiss={onDismiss} />
```

Comment:
i'm curious, couldn't we make `onDismiss` optional, and create the `dismissable` flag inside the `Pill` component `const dismissable = !!onDismiss`? b/c it looks a little redundant to apply both dismissable and a onDismiss func

Reply:
Yep great idea, I'll do that